common/static/common/js/discussion/views/discussion_thread_show_view.js

```diff
@@ -38,32 +38,34 @@
 
             DiscussionThreadShowView.prototype.renderTemplate = function() {
                 var context;
-                this.template = _.template($("#thread-show-template").html());
                 context = $.extend({
                     mode: this.mode,
                     flagged: this.model.isFlagged(),
                     author_display: this.getAuthorDisplay(),
                     cid: this.model.cid,
                     readOnly: $('.discussion-module').data('read-only')
                 }, this.model.attributes);
-                return this.template(context);
+                return edx.HtmlUtils.template($("#thread-show-template").html())(context);
             };
 
             DiscussionThreadShowView.prototype.render = function() {
-                this.$el.html(this.renderTemplate());
+                edx.HtmlUtils.setHtml(
+                    this.$el,
+                    this.renderTemplate()
+                );
                 this.delegateEvents();
                 this.renderAttrs();
                 this.$("span.timeago").timeago();
                 this.convertMath();
-                this.highlight(this.$(".post-body"));
-                this.highlight(this.$("h1,h3"));
+                this.$(".post-body");
+                this.$("h1,h3");
                 return this;
             };
 
             DiscussionThreadShowView.prototype.convertMath = function() {
-                var element;
-                element = this.$(".post-body");
-                element.html(DiscussionUtil.postMathJaxProcessor(DiscussionUtil.markdownWithHighlight(element.text())));
```

Comment:
@e-kolpakov: I still think that `alreadyEscaped` code is odd, and means something went a bit awry.  I think part of the problem is in this line that went away.  This used to call `element.text()`, and you now have it as 'element.html()' in convertMath().  I'm not sure if that is the only change needed, or if different calls were made in different contexts?